core/Dockerfile

```diff
@@ -36,6 +37,7 @@ RUN /home/pi/services/install-services.sh
 
 # Copy binaries from downloadBinaries to this stage
 COPY --from=downloadBinaries \
+    /usr/bin/ardupilot_fw_uploader.py \
```

Comment:
That will probably also help with @Williangalvani suggestion for flashing process through the REST API.

Reply:
It doesn't change because the uploader tool does not exposes the progress internally. It's a tool mainly designed to be used as a final program, not a library.

We can modify it to expose its internal state and allow proper usage as a library, but I propose we do that in a second moment.

Reply:
https://github.com/ArduPilot/ardupilot/blob/master/Tools/scripts/uploader.py#L176
It exposes via the uploader class.

Reply:
In which line? From what I can see [here](https://github.com/ArduPilot/ardupilot/blob/421faa0ada08ba4af395ff8a83291617d876b043/Tools/scripts/uploader.py#L507) the progress is not a class attribute.

Reply:
We could improve it in the future so, but good to know.

Reply:
Yes.
There are some good changes that could be made on uploader.py to make it more friendly to be used as a library.
Currently to use it in a very fashioned way to make it work. Something like this:

```python
def upload(self, filepath):
    fw = firmware(filepath)
    for i in range(10):
        print(f"Attempting to upload firmware... ({i+1}/10)")
        try:
            up = uploader(
                portname=self.autopilot_port,
                baudrate_bootloader=self.autopilot_bootloader_baud,
                baudrate_flightstack=self.autopilot_flightstack_bauds,
            )
        except Exception as e:
            print("Exception creating uploader: %s" % str(e))
            time.sleep(0.05)
            continue
        if not find_bootloader(up, self.autopilot_port):
            continue
        try:
            up.upload(fw)
            print("Sucefully uploaded new firmware. ")
            break
        except RuntimeError as e:
            print(f"ERROR: {e.args}\n")
        except IOError as e:
            print(f"ERROR: {e.args}\n")
        finally:
            up.close()
```